src/gobexport/test.py

```diff
@@ -392,84 +399,121 @@ def _check_file(check, filename, stats):
     return total_result
 
 
-def _get_analysis(obj_info, obj, check=None):
+def _get_analysis(obj_info: dict, obj: bytes, check: Optional[dict] = None):
     """
     Get statistics for the given file (object)
 
     :param obj_info: meta information
     :param obj: contents
     :return:
     """
-    ENCODING = 'utf-8'
-    check = check or {}
+    base_analysis = {
+        "age_hours": (
+            (datetime.datetime.now() - dateutil.parser.parse(obj_info["last_modified"])).total_seconds() / 3600
+        ),
+        "bytes": obj_info["bytes"],
+        "first_bytes": hashlib.md5(obj[:10000]).hexdigest()
+    }
 
-    last_modified = obj_info["last_modified"]
-    age = datetime.datetime.now() - dateutil.parser.parse(last_modified)
-    age_hours = age.total_seconds() / (60 * 60)
+    if obj_info['content_type'] in ("plain/text", "text/csv", "application/x-ndjson") and obj_info["bytes"] > 0:
+        lines, stats = _get_line_stats_from_bytes(obj)
 
-    bytes = obj_info["bytes"]
+        base_analysis.update(
+            _get_first_lines_stats(lines) | stats | _check_csv(lines, obj_info, check)
+        )
 
-    first_bytes = hashlib.md5(obj[:10000]).hexdigest()
+    return base_analysis
 
-    base_analysis = {
-        "age_hours": age_hours,
-        "bytes": bytes,
-        "first_bytes": first_bytes
-    }
 
-    if obj_info['content_type'] not in ["plain/text", "text/csv", "application/x-ndjson"] or bytes == 0:
-        return base_analysis
+def _get_line_stats_from_bytes(obj: bytes) -> tuple[list[str], dict[str, numbers.Number]]:  # noqa: C901
+    chars, digits, alphas, spaces, lowers, uppers, empty_lines, total_line = [0] * 8
+    min_line = -1
+    max_line = -1
 
-    content = obj.decode(ENCODING)
-    chars = len(content)
+    def _process_line(line: str) -> str:
+        nonlocal chars, digits, alphas, spaces, lowers, uppers, empty_lines, max_line, min_line, total_line
 
-    lines = content.split('\n')
+        # all chars
+        chars += len(line)
 
-    cols = _check_csv(lines, obj_info, check)
+        for char in line:
```

Comment:
This for-loop hurts my eyes with all the nested if/elses. What has changed compared to the original, much easier readable code?